### PR TITLE
[Datafactory] Adding getFeatureValue API in factory scope

### PR DESCRIPTION
**Pull request by hvermis**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5746 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5746/commits/b703a3fda59fa059678952c58eeafdcbff1c9e94

**Comment by anuchandy** [comment]
@ravbhatnagar addition of API in an existing api-version, please review.

**Comment by hvermis** [comment]
@ravbhatnagar , @anuchandy This API already exists in subscriptions scope, we are adding it in factory scope.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json#L159

**Review by anuchandy — COMMENTED** [review]
@hvermis it seems there are model validation errors in existing examples. Refer https://dev.azure.com/azure-sdk/public/_build/results?buildId=16345 expand "ModelValidation Job" to see the examples for which validation is failing.

**Comment by hvermis** [comment]
@anuchandy It is still running. Did you restart it?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comment

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -204,6 +204,57 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/getFeatureValue": {
+      "post": {
```
[the remark this answers is not in the document]
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: All our APIs are following guideline where Get is used only on items modeled as resources, and the rest are actions with POST. FeatureValues are not modeled as resources, as we don't intend to allow CRUD operations on them, and no list by parent. Validators report these as errors.

**Comment by anuchandy** [comment]
Requesting for reviewer reassign since I'm OOF

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: @KrisBash I replied to your comment.

**Comment by hvermis** [comment]
@anuchandy or next reviewer, I'm trying to fix the violation errors. The main error is that our examples don't have 304 status code. How do I add them for the same API? The request and the response are slightly different from 200 status code. Should I add them into the same file or a separate file?

**Comment by anuchandy** [comment]
@hvermis you can use the same example to specify both status codes response: https://github.com/Azure/azure-rest-api-specs/blob/0060c894bee5fc074fec73d9bbf9f5682575216b/specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/CreateAScaleSetWithPasswordAuthentication.json#L66

**Comment by hvermis** [comment]
@anuchandy in our case the request is different too. In the example you shared the request payload is the same

**Comment by anuchandy** [comment]
@hvermis Looking at the model validation error - I guess this is about `Triggers_Get` API that optionally allows setting `If-None-Match` header. 

In the current [example RESPONSE](https://github.com/Azure/azure-rest-api-specs/blob/2130f16f3b4eac241c8a50566a5dc4498d081f7b/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/Triggers_Get.json#L50), the e-tag is `0a008dd4-0000-0000-0000-5b245c6f0000`. In the same example, we could modify the REQUEST to include `If-None-Match` with a value DIFFERENT from `0a008dd4-0000-0000-0000-5b245c6f0000`and add 304 response with empty body. This way the example cover both cases:
 (1). if the specified request e-tag matches then `304 {}` 
(2). if not `200: {...body..}`.

What do you think?

**Comment by hvermis** [comment]
@anuchandy yes, I already have the examples, I just don't know what format the validation tool is expecting?
Like this in the same file?
{
  "parameters": {
    "subscriptionId": "12345678-1234-1234-1234-12345678abc",
    "resourceGroupName": "exampleResourceGroup",
    "factoryName": "exampleResourceGroup",
    "integrationRuntimeName": "exampleIntegrationRuntime",
    "ifNoneMatch": null,
    "api-version": "2018-06-01"
  },
  "responses": {
    "200": {
      "headers": {
        "Date": "Thu, 11 Apr 2019 00:17:58 GMT",
        "X-Content-Type-Options": "nosniff",
        "x-ms-ratelimit-remaining-subscription-reads": "11997",
        "x-ms-request-id": "9fe52bc2-8aeb-47fe-8bf9-3b6657455278",
        "x-ms-correlation-request-id": "9fe52bc2-8aeb-47fe-8bf9-3b6657455278"
      },
      "body": {
        "id": "/subscriptions/12345678-1234-1234-1234-12345678abc/resourceGroups/exampleResourceGroup/providers/Microsoft.DataFactory/factories/exampleResourceGroup/integrationruntimes/exampleIntegrationRuntime",
        "name": "exampleIntegrationRuntime",
        "type": "Microsoft.DataFactory/factories/integrationruntimes",
        "properties": {
          "type": "SelfHosted",
          "description": "A selfhosted integration runtime"
        },
        "etag": "0100e789-0000-0000-0000-5cae87b10000"
      }
    }
  }
}
{
  "parameters": {
    "subscriptionId": "12345678-1234-1234-1234-12345678abc",
    "resourceGroupName": "exampleResourceGroup",
    "factoryName": "exampleResourceGroup",
    "integrationRuntimeName": "exampleIntegrationRuntime",
    "ifNoneMatch": "0100e789-0000-0000-0000-5cae87b10000",
    "api-version": "2018-06-01"
  },
  "responses": {
    "304": {
      "headers": {
        "Date": "Thu, 11 Apr 2019 00:18:20 GMT",
        "X-Content-Type-Options": "nosniff",
        "x-ms-ratelimit-remaining-subscription-reads": "11995",
        "x-ms-request-id": "c985171d-03e6-4400-8567-e8868ce91bde",
        "x-ms-correlation-request-id": "c985171d-03e6-4400-8567-e8868ce91bde"
      },
      "body": null
    }
  }
}

**Review by veronicagg — CHANGES_REQUESTED** [review]
@hvermis regarding the model validation failure - here's an example of specifying discriminator in swagger: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json#L12, and below that definition you can see the "sub type" which inherits from "LinkedService". And this is how the example looks like for LinkedServices_Create https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/LinkedServices_Create.json to pass model validation.
It appears that the example included for the [failure](https://dev.azure.com/azure-sdk/public/_build/results?buildId=21080&view=logs&jobId=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&taskId=43324f1f-6724-5d61-785f-f0ac9fd256a1&lineStart=47&lineEnd=48&colStart=1&colEnd=1) does not include the type for the discriminator in the return values. 
It also appears that this model validator issue is not being introduced by this PR. So, I would recommend to follow up, but you could fix this one in a subsequent PR, as it's not related to this one. Let me know what you prefer.

**Comment by hvermis** [comment]
@veronicagg It seems I can't fix the last validation error in this PR. Can you please go ahead and merge my PR? I will follow up with my teammates to get SSIS object metadata models in swagger fixed and examples updated.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/IntegrationRuntimeObjectMetadata_Get.json [hunk 2]
```diff
@@ -39,390 +39,6 @@
             "id": 4,
             "name": "这是文件夹",
             "description": ""
```
Comment (zhangyd2015) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/IntegrationRuntimeObjectMetadata_Get.json]: This example needs to be re-captured after Xiao's fix.

**Review by zhangyd2015 — APPROVED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/IntegrationRuntimeObjectMetadata_Get.json (hunk 2 above)
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/IntegrationRuntimeObjectMetadata_Get.json]: It should be done in Xiao's PR.

**Comment by veronicagg** [comment]
@hvermis since the model validation failures are not related, yes. ARM team hasn't signed off yet.

**Review by KrisBash — APPROVED** [review]
Signing off

**Comment by hvermis** [comment]
@veronicagg  Can you please merge this PR?

**Review by veronicagg — APPROVED** [review]

**Pull request merged**